File: Second Year (2013-2014)/OMGL7/TP3 - Website in JSP/src/db/ClientDAO.java
package db;

import java.sql.*;
/**
* @author Andriamilanto Tompoariniaina - Canno Camille - Info2B
*/

/**
* Classe pour les objets DAO de Client
*/
public class ClientDAO {
//Methodes
	/**
	*Constructeur
	*/
	public ClientDAO() {}

	/**
	*Pour retrouver tous les tuples
	*/
	public ResultSet findAll() {
		Statement stat = null;
		String query = "";
		ResultSet ret = null;

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Client;";

			//Le resultat a retourner
			ret = stat.executeQuery(query);
		}
		catch (SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		//Retourne l'execution de la requete sous la forme d'un objet ResultSet
		return ret;
	}

	/**
	*Permet de retrouver juste un tuple
	*@param id idClient du client a retrouver
	*/
	public Client find(String login) {
		Statement stat = null;
		String query = "";
		Client ret = new Client();

		try {
			//Recuperation de la connexion
			Connection con = MySQLConnection.getInstance().getConnection();

			//Preparation de la requete en ligne
			stat = con.createStatement();

			//Preparation de la requete
			query = "SELECT * FROM Client WHERE Login = " + login + ";";

			//Retourne l'execution de la requete sous la forme d'un objet ResultSet
			ResultSet result = stat.executeQuery(query);

			//Si le resultat est bon, prends la premiere ligne
			if (result.first()) {
				ret.init(login, result.getString(2), result.getString(3), result.getString(4), result.getDate(5), result.getString(6), result.getString(7), result.getInt(8), result.getString(9), result.getInt(10));
			}
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}

		return ret;
	}


	/**
    *Methode qui permet d'inserer un tuple
    *@param tuple Objet de type Client a inserer
    */
    public void insert(Client tuple) {
    	Statement stat = null;
        String query = "";

        //Recuperation des attributs de l'objet Client
        String login = tuple.getLogin();
        String passwd = tuple.getPassword();
        String nom = tuple.getNom();
        String prenom = tuple.getPrenom();
        Date ddn = tuple.getDateDeNaissance();
        String email = tuple.getEmail();
        String adresse = tuple.getAdresse();
        int codePostal = tuple.getCodePostal();
        String ville = tuple.getVille();
        int idP = tuple.getIdPays();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "INSERT INTO Client VALUES(" + login + "," + passwd + "," + nom + "," + prenom + "," + ddn + "," + email + "," + adresse + "," + codePostal + "," + ville + "," + idP + ");";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de faire la mise a jour d'un tuple
    *@param tuple Objet nouveau a mettre a la place de l'ancien
    */
    public void update(Client tuple) {
    	Statement stat = null;
        String query = "";

       	//Recuperation des attributs de l'objet Client
        String login = tuple.getLogin();
        String passwd = tuple.getPassword();
        String nom = tuple.getNom();
        String prenom = tuple.getPrenom();
        Date ddn = tuple.getDateDeNaissance();
        String email = tuple.getEmail();
        String adresse = tuple.getAdresse();
        int codePostal = tuple.getCodePostal();
        String ville = tuple.getVille();
        int idP = tuple.getIdPays();

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "UPDATE Client SET Password = " + passwd + ", Nom = " + nom + ", Prenom = " + prenom + ", DateDeNaissance = " + ddn + ", Email = " + email + ", Adresse = " + adresse + ", CodePostal = " + codePostal + ", Ville = " + ville + ", IdPays = " + idP + " WHERE Login = " + login + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }


    /**
    * Permet de supprimer un tuple
    *@param idclient id du tuple a supprimer
    */
    public void delete(String login) {
    	Statement stat = null;
        String query = "";

        try {
	    	//Recuperation de la connexion
	        Connection con = MySQLConnection.getInstance().getConnection();

	        //Preparation de la requete
	        query = "DELETE FROM Client WHERE Login = " + login + ";";

	        //Execute la requête
			stat.executeQuery(query);
		}
		catch(SQLException e) {
			System.out.println("ERREUR: " + e.getMessage());
		}
    }
}

File: Second Year (2013-2014)/OMGL7/TP2 - Website in PHP/musiconline/musiconline/db/ClientDao.php
<?php
/* Authors: ANDRIAMILANTO Tompoariniaina && CANNO Camille
* Groupe: B1
* File: ClientDao.php
*/


import ('phpens.framework.Application');
import ('phpens.framework.db.DataObject');
import ('phpens.framework.db.DataAccessObject');
import ('musiconline.db.Client');


class ClientDao implements DataAccessObject {
  //Attribut instance de DataAccessObject
  private static $dao;

  //Constructeur vide
  private function __construct() { }

  //Retourne l'instance de ClientDAO
  public final static function getInstance() {

    //Vérifie d'abord si l'instance est correcte ou non
    if(!isset(self::$dao)) {

      //Si non correct, en créer une
      self::$dao= new ClientDao();
    }

    //Retourne finallement l'instance, sinon retourne une instance vide
    return self::$dao;
  }


  //Retourne toute la table
  public final function findAll() {

    //Récupère la méthode de connexion au SGBD
    $dbc = Application::getInstance()->getDBConnection();

    //Prépare la requête
    $query = "select * from Client";

    //Requête en dur, prends la BDD et y exécute la requête
    $stmt = $dbc->query($query);

    //Récupère le résultat sous forme de tableau d'instances de classes
    $results = $stmt->fetchALL(PDO::FETCH_CLASS, 'Client');

    //Retourne le résultat
    return $results;
   }


  //Permet d'insérer un tuple
  public final function insert(DataObject $do) {

    //Vérifie que l'objet entré est bien une instance de Client
    if($do instanceof Client){

      //Récupère la méthode de connexion au SGBD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "insert into Client(Login,Password,Nom,Prenom,DateDeNaissance,Email,Adresse,CodePostal,Ville,IdPays) values (:Login,:Password,:Nom,:Prenom,:DateDeNaissance,:Email,:Adresse,:CodePostal,:Ville,:IdPays)";

      //Prépare la requête en dûr
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':Login',$do->getLOgin(),PDO::PARAM_STR);
      $stmt->bindValue(':Password',$do->getPAssword(),PDO::PARAM_STR);
      $stmt->bindValue(':Nom',$do->getNOm(),PDO::PARAM_STR);
      $stmt->bindValue(':Prenom',$do->getPRenom(),PDO::PARAM_STR);
      $stmt->bindValue(':DateDeNaissance',$do->getDAteDeNaissance(),PDO::PARAM_STR);
      $stmt->bindValue(':Email',$do->getEMail(),PDO::PARAM_STR);
      $stmt->bindValue(':Adresse',$do->getADresse(),PDO::PARAM_STR);
      $stmt->bindValue(':CodePostal',$do->getCOdePostal(),PDO::PARAM_STR);
      $stmt->bindValue(':Ville',$do->getVIlle(),PDO::PARAM_STR);
      $stmt->bindValue(':IdPays',$do->getIDPays(),PDO::PARAM_STR);
      
      //Exécute la requête finale
      $stmt->execute();
   }
 }


  //Méthode qui permet de mettre à jour un tuple
  public function update(DataObject $obj) {

    //Vérifie que l'objet entré est bien une instance de Client
    if($obj instanceof Client){

      //Récupère la méthode de connexion au SGBD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "UPDATE Client SET Password = :Password, Nom = :Nom, Prenom = :Prenom, DateDeNaissance = :DateDeNaissance, Email = :Email, Adresse = :Adresse, CodePostal = :CodePostal, Ville = :Ville, IdPays = :IdPays WHERE Login = :Login";

      //Prépare la requête en dûr
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':Login',$obj->getLOgin(),PDO::PARAM_STR);
      $stmt->bindValue(':Password',$obj->getPAssword(),PDO::PARAM_STR);
      $stmt->bindValue(':Nom',$obj->getNOm(),PDO::PARAM_STR);
      $stmt->bindValue(':Prenom',$obj->getPRenom(),PDO::PARAM_STR);
      $stmt->bindValue(':DateDeNaissance',$obj->getDAteDeNaissance(),PDO::PARAM_STR);
      $stmt->bindValue(':Email',$obj->getEMail(),PDO::PARAM_STR);
      $stmt->bindValue(':Adresse',$obj->getADresse(),PDO::PARAM_STR);
      $stmt->bindValue(':CodePostal',$obj->getCOdePostal(),PDO::PARAM_STR);
      $stmt->bindValue(':Ville',$obj->getVIlle(),PDO::PARAM_STR);
      $stmt->bindValue(':IdPays',$obj->getIDPays(),PDO::PARAM_STR);
      
      //Exécute la requête finale
      $stmt->execute();
  }
  }


  //Méthode permettant de supprimer un tuple
  public function delete(DataObject $obj) {

    //Vérifie si l'objet entré en paramètre est instance de Album
    if($obj instanceof Client) {

      //Récupère la méthode de connection à la BDD
      $dbc = Application::getInstance()->getDBConnection();

      //Prépare la requête SQL
      $query = "DELETE FROM Client WHERE Login = :Login";
    
      //Prépare la requête en dur avec la BDD et la requête correspondante
      $stmt = $dbc->prepare($query);

      //Remplace les paramètres
      $stmt->bindValue(':Login',$obj->getLOgin(),PDO::PARAM_STR);
   
      //Exécute finallement la requête
      $stmt->execute();
  }
}
}
?>
